src/som/interpreter/actors/EventualMessage.java

```diff
@@ -73,8 +81,15 @@ public SourceSection getTargetSourceSection() {
 
   public long serialize(final SnapshotBuffer sb) {
     ReceivedRootNode rm = (ReceivedRootNode) this.onReceive.getRootNode();
-    // Not sure if this is optimized, worst case need to duplicate this for all messages
-    return rm.getSerializer().execute(this, sb);
+
+    if (VmSettings.TEST_SNAPSHOTS || VmSettings.TEST_SERIALIZE_ALL
+        || sb.getSnapshotVersion() > this.getMessageId()) {
```

Comment:
```suggestion
        || sb.needsToBeSnapshot(getMessageId())) {
```

This would be one way to encapsulate the logic.

I am also rather surprised by all the `||` in the expression. Shouldn't it be first an `&&` and then in parentheses a `||`? I am just guess, but, an all `||` looks odd.

Reply:
I probably should rename TEST_SNAPSHOTS to TEST_SERIALIZATION or something. It is used for the serialization test, i.e. serialization of individual object(graphs). For that test we don't trigger normal snapshots, as a result all snapshot versions are 0 and we need the ||.